internal/commands/schema/models_inputs.go

```diff
@@ -0,0 +1,155 @@
+package schema
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+const (
+	flagModelsFlat      = "flat"
+	flagModelsFlatUsage = "the generated data models will be grouped (with their imports) in one code block"
+
+	flagModelsLanguage      = "language"
+	flagModelsLanguageShort = "l"
+	flagModelsLanguageUsage = "specify the language to generate schema data models for"
+
+	flagModelsNoImports      = "no-imports"
+	flagModelsNoImportsUsage = "include to omit import statements from the generated data model code"
+
+	flagModelsName      = "name"
+	flagModelsNameUsage = "use (as an array or csv flag) to filter for matched schema object names"
+)
+
+type modelsInputs struct {
+	cli.ProjectInputs
+	Flat      bool
+	Language  language
+	NoImports bool
+	Names     []string
+	nameSet   map[string]struct{}
+}
+
+func (i *modelsInputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if i.Language == "" {
+		options, typesByOption := make([]string, len(allLanguages)), map[interface{}]language{}
+		for i, l := range allLanguages {
+			o := languageDisplay(l)
+			options[i] = o
+			typesByOption[o] = l
+		}
+
+		var lang string
+		if err := ui.AskOne(&lang, &survey.Select{
+			Message: "Select the language you would like to generate data models in",
+			Options: options,
+		}); err != nil {
+			return err
+		}
+		i.Language = typesByOption[lang]
+
+		if !i.NoImports {
+			var noImports bool
+			if err := ui.AskOne(&noImports, &survey.Confirm{Message: "Would you like to omit imports?"}); err != nil {
+				return err
+			}
+			i.NoImports = noImports
+		}
+
+		if !i.Flat {
+			var flat bool
+			if err := ui.AskOne(&flat, &survey.Confirm{Message: "Would you like group all generated data models together?"}); err != nil {
+				return err
+			}
+			i.Flat = flat
+		}
```

Comment:
[q] Going the route of always asking if flag isn't set?

Reply:
haha yeah, based off the feedback from our past bug bash, it felt like users appreciated more prompts/less flags.  fwiw, this "prompt if not set" behavior still will only occur if the user omits `--language` flag ... if they were to set `--language` but not set `--flat` or `--no-imports`, the prompts would still not pop up